backend/otel/extract.go

```diff
@@ -127,169 +132,156 @@ func extractFields(ctx context.Context, params extractFieldsParams) (extractedFi
 		}
 	}
 
-	attrs := mergeMaps(
+	originalAttrs := mergeMaps(
 		resourceAttributes,
 		spanAttributes,
 		eventAttributes,
 		scopeAttributes,
 		logAttributes,
 	)
 
-	// process potential syslog message
-	if len(fields.logBody) > 0 && fields.logBody[0] == '<' {
-		extractSyslog(&fields, attrs)
+	if val, ok := originalAttrs[highlight.DeprecatedSourceAttribute]; ok {
+		if val == modelInputs.LogSourceFrontend.String() {
+			fields.source = modelInputs.LogSourceFrontend
+		}
+		delete(originalAttrs, highlight.DeprecatedSourceAttribute)
 	}
 
-	if val, ok := attrs[highlight.DeprecatedProjectIDAttribute]; ok {
-		fields.projectID = val.(string)
-		delete(attrs, highlight.DeprecatedProjectIDAttribute)
+	if val, ok := originalAttrs[highlight.SourceAttribute]; ok {
+		if val == modelInputs.LogSourceFrontend.String() {
+			fields.source = modelInputs.LogSourceFrontend
+		}
+		delete(originalAttrs, highlight.SourceAttribute)
 	}
 
-	if val, ok := attrs[highlight.ProjectIDAttribute]; ok {
-		fields.projectID = val.(string)
-		delete(attrs, highlight.ProjectIDAttribute)
+	for k, v := range originalAttrs {
+		prefixes := []string{}
+		if fields.source == modelInputs.LogSourceFrontend {
```

Comment:
I moved this logic up the chain (see how it was down below) such that we can chop off `service.*` before we try extracting it into the top level `serviceName`